doc/admin/dictionary.rst

```diff
@@ -0,0 +1,81 @@
+client and the KDC.  Online dictionary attacks can be mitigated using
+:ref:`account lockout <lockout>`.  This measure is not totally
+satisfactory, as it makes it easy for an attacker to deny access to a
+client principal.
+
+An offline dictionary attack is performed by obtaining a ciphertext
+generated using the password-derived key, and trying each password
+against the ciphertext.  This category of attack is invisible to the
+KDC and can be performed much faster than an online attack.  The
+attack will generally take much longer with more recent encryption
+types (particularly the ones based on AES), because those encryption
+types use a much more expensive string-to-key function.  However, the
+best defense is to deny the attacker access to a useful ciphertext.
+The required defensive measures depend on the attacker's level of
+network access.
+
+An off-path attacker has no access to packets sent between legitimate
+users and the KDC.  An off-path attacker could gain access to an
+attackable ciphertext either by making an AS request for a client
+principal which does not have the **+requires_preauth** flag, or by
+making a TGS request (after authenticating as a different user) for a
+server principal which does not have the **-allow_svr** flag.  To
+address off-path attackers, a KDC administrator should set those flags
+on principals with password-derived keys::
+
+    kadmin: add_principal +requires_preauth -allow_svr princname
+
+An attacker with passive network access (one who can monitor packets
+sent between legitimate users and the KDC, but cannot change them or
+insert their own packets) can gain access to an attackable ciphertext
+by observing an authentication by a user using the most common form of
+preauthentication, encrypted timestamp.  Any of the following methods
+can prevent dictionary attacks by attackers with passive network
+access:
+
+* Enabling :ref:`SPAKE preauthentication <spake>` (added in release
+  1.17) on the KDC, and ensuring that all clients are able to support
+  it.
+
+* Using an :ref:`HTTPS proxy <https>` for communication with the KDC.
+
+* Using FAST, protecting the initial authentication with either a
+  random key (such as a host key) or with :ref:`anonymous PKINIT
+  <anonymous_pkinit>`.
+
+An attacker with active network access (one who can inject or modify
+packets sent between legitimate users and the KDC) can try to fool the
+client software into sending an attackable ciphertext using an
+encryption type and salt string of the attacker's choosing.  Any of the
+following methods can prevent dictionary attacks by active attackers:
+
+* Enabling SPAKE preauthentication and setting the
+  **disable_encrypted_timestamp** variable to ``true`` in the
+  :ref:`realms` subsection of the client configuration.
+
+* Using an HTTPS proxy, configured in the client's krb5.conf realm
+  configuration.  Using :ref:`KDC discovery <kdc_discovery>` may be
+  defeated with DNS spoofing.
```

Comment:
Should probably expound a little bit more on the second sentence, about who is defeating what.

Reply:
Done, along with a few other adjustments.